dvc/analytics.py

```diff
@@ -1,256 +1,124 @@
-from dvc.utils import env2bool
-from dvc.utils.compat import str
+from dvc.config import Config, to_bool
+from dvc.exceptions import NotDvcRepoError
+from dvc.lock import Lock, LockError
+from dvc.repo import Repo
+from dvc.scm import SCM
+from dvc.utils import env2bool, is_binary, makedirs
+from dvc.utils.compat import str, FileNotFoundError
 
 
 logger = logging.getLogger(__name__)
 
 
-class Analytics(object):
-    """Class for collecting and sending usage analytics.
-
-    Args:
-        info (dict): optional existing analytics report.
+def collect(args=None, return_code=None):
     """
+    Query the system to fill a report.
+    """
+    return {
+        "cmd_class": args.func.__name__ if hasattr(args, "func") else None,
+        "cmd_return_code": return_code,
+        "dvc_version": __version__,
+        "is_binary": is_binary(),
+        "scm_class": scm_in_use(),
+        "system_info": system_info(),
+        "user_id": find_or_create_user_id(),
+    }
 
-    URL = "https://analytics.dvc.org"
-    TIMEOUT_POST = 5
-
-    USER_ID_FILE = "user_id"
-
-    PARAM_DVC_VERSION = "dvc_version"
-    PARAM_USER_ID = "user_id"
-    PARAM_SYSTEM_INFO = "system_info"
-
-    PARAM_OS = "os"
-
-    PARAM_WINDOWS_VERSION_MAJOR = "windows_version_major"
-    PARAM_WINDOWS_VERSION_MINOR = "windows_version_minor"
-    PARAM_WINDOWS_VERSION_BUILD = "windows_version_build"
-    PARAM_WINDOWS_VERSION_SERVICE_PACK = "windows_version_service_pack"
-
-    PARAM_MAC_VERSION = "mac_version"
-
-    PARAM_LINUX_DISTRO = "linux_distro"
-    PARAM_LINUX_DISTRO_VERSION = "linux_distro_version"
-    PARAM_LINUX_DISTRO_LIKE = "linux_distro_like"
-
-    PARAM_SCM_CLASS = "scm_class"
-    PARAM_IS_BINARY = "is_binary"
-    PARAM_CMD_CLASS = "cmd_class"
-    PARAM_CMD_RETURN_CODE = "cmd_return_code"
-
-    def __init__(self, info=None):
-        from dvc.config import Config
-        from dvc.lock import Lock
-
-        if info is None:
-            info = {}
-
-        self.info = info
-
-        cdir = Config.get_global_config_dir()
-        try:
-            os.makedirs(cdir)
-        except OSError as exc:
-            if exc.errno != errno.EEXIST:
-                raise
 
-        self.user_id_file = os.path.join(cdir, self.USER_ID_FILE)
-        self.user_id_file_lock = Lock(self.user_id_file + ".lock")
+def is_enabled():
+    if env2bool("DVC_TEST"):
+        return False
 
-    @staticmethod
-    def load(path):
-        """Loads analytics report from json file specified by path.
+    enabled = to_bool(
+        Config(validate=False)
+        .config.get(Config.SECTION_CORE, {})
+        .get(Config.SECTION_CORE_ANALYTICS, "true")
+    )
 
-        Args:
-            path (str): path to json file with analytics report.
-        """
-        with open(path, "r") as fobj:
-            analytics = Analytics(info=json.load(fobj))
-        os.unlink(path)
-        return analytics
+    logger.debug("Analytics is {}enabled.".format("" if enabled else "dis"))
```

Comment:
`disenabled`? 🙂

Reply:
I don't know what I was thinking about :rofl:

Reply:
probably thinking in Spanish :disappointed: